### PR TITLE
Adds ThemeController

### PR DESCRIPTION
**Pull request by matuella**

### CODE REVIEW

File: lib/application/theme/memo_theme_data.dart [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+enum MemoTheme { classic }
+
+/// Wraps all fundamental swatches for a `MemoTheme`
+///
+/// The only purpose of having this customized set of theme-related properties is due to the fact that the material's
+/// `ThemeData` doesn't support custom properties. While there are workarounds, they don't quite work for applications
+/// that require multiple sub-themes.
+///
+/// There is an [open issue](https://github.com/flutter/flutter/issues/31522) in flutter's frameowkr, tracking this
```
Comment (ggirotto) [inline lib/application/theme/memo_theme_data.dart:11]: frameowkr*

File: lib/application/theme/memo_theme_colors.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      600: Color(0xFF6E57B0),
+      700: Color(0xFF594593),
+      800: Color(0xFF453475),
+      900: Color(0xFF322558),
+    },
+  );
+}
+
+MaterialColor buildClassicNeutralSwatch() {
+  const _primaryNeutral = Color(0xFF7A748E);
```
Comment (ggirotto) [inline lib/application/theme/memo_theme_colors.dart:49]: Above swatches use just `_primary`, couldn't this one follow the same naming pattern?

File: lib/application/theme/theme_controller.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:memo/application/constants/dimensions.dart' as dimens;
+import 'package:memo/application/theme/material_theme_data.dart' as material_theme;
+import 'package:memo/application/theme/memo_theme_colors.dart' as colors;
+import 'package:memo/application/theme/memo_theme_data.dart';
+
+final themeProvider = StateNotifierProvider<ThemeController>((ref) => ThemeController());
```
Comment (ggirotto) [inline lib/application/theme/theme_controller.dart:9]: `ref` is not used in its scope

Comment (ggirotto) [reply lib/application/theme/theme_controller.dart:9]: Also, isn't lint complaining about this?

**Review by ggirotto — CHANGES_REQUESTED** [review]

File: lib/application/theme/theme_controller.dart (hunk 3 above)
Comment (matuella) [reply lib/application/theme/theme_controller.dart:9]: I don't think dart has a check for unused parameters inside a particular closure

**Review by ggirotto — APPROVED** [review]

**Pull request merged**